CONTRIBUTING.md

```diff
@@ -170,6 +170,16 @@ for coding advice). The code must adhere to the following robustness principles
 are important for software that runs autonomously and continuously without direct
 interaction with a human (such as this Collector).
 
+### Method naming
+
+To keep naming patterns consistent across the project, naming patterns are enforced to make intent clear by:
+
+- Any method that allocates memory onto the heap must use the name prefix `New`
+  - For example `NewKinesisExporter` will return a kinesis exporter using the zero values 
+    or values provided by the provided arguments
+- Any method that references a static / global object must use the name prefix `Default`
+  - For example `DefaultKinesisConfig` returns a copy of an object that is configured with non zero values 
```

Comment:
I think this is what we discussed in https://github.com/open-telemetry/opentelemetry-collector/issues/4471.

My understanding was that:
* If func (e.g. `func NewDefaultKinesisConfig() KinesisConfig{...}`) we use `New` even if not heap allocated.
* If var (e.g. `var DefaultKinesisConfig = KinesisConfig{...}`) we don't use `New` as prefix.

Reply:
You're right, let me quickly fix this :)